File: quiz_app_flutter/lib/domain/usecase/result/saveresultdata_usecase.dart
import 'dart:convert';

import 'package:quiz_app_flutter/data/cache/cache_manager.dart';
import 'package:quiz_app_flutter/domain/entities/result/result_data.dart';

class SaveResultDataUseCase {
  final CacheManager cacheManager;

  SaveResultDataUseCase(this.cacheManager);

  Stream<void> call(String key, ResultData result) async* {
    final jsonString = json.encode(result.toJson());
    await cacheManager.saveResult(key, jsonString);
    yield null; // Emit null to indicate success (like Unit in Kotlin)
  }
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/domain/usecase/result/SaveResultDataUseCase.kt
package com.tnm.quizappkmp.domain.usecase.result

import com.tnm.quizappkmp.data.cache.CacheManager
import com.tnm.quizappkmp.domain.model.result.ResultData
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import kotlinx.serialization.json.Json

class SaveResultDataUseCase(
    private val cacheManager: CacheManager,
    private val json: Json
) {
    operator fun invoke(key: String, result: ResultData): Flow<Unit> = flow {
        cacheManager.saveResult(key, json.encodeToString(result))
        emit(Unit) // Emit Unit to indicate success
    }
}